master/job.go

```diff
@@ -0,0 +1,91 @@
+package master
+
+import (
+	"github.com/aaronang/cong-the-ripper/lib"
+)
+
+// NOTE: consider using math/big for some of the operations in this package
+
+type job struct {
+	lib.Job
+	id    int
+	tasks []lib.Task
+}
+
+// SplitJob attempts to split a cracking job into equal sized tasks regardless of the job
+// the taskSize represents the number of brute force iterations
+func SplitJob(job *job, taskSize int) []lib.Task {
+	var tasks []lib.Task
+	cands, lens := chunkCharSet(job.CharSet, job.KeyLen, taskSize)
+	for i := range cands {
+		tasks = append(tasks, lib.Task{
+			Job:     job.Job,
+			JobID:   job.id,
+			ID:      i,
+			Start:   cands[i],
+			TaskLen: lens[i]})
+	}
+	return tasks
+}
+
+// chunkCharSet takes a character set and the required length l and splits to chunks of size n
+func chunkCharSet(charset lib.CharSet, l, n int) ([][]byte, []int) {
+	cand := initialCandidate(charset, l)
+	var cands [][]byte
+	var ints []int
+	for {
+		newComb, carry := nthCandidateFrom(charset, n, cand)
+		if carry == 0 {
+			ints = append(ints, n)
+			cands = append(cands, cand)
+		} else {
+			// this part is not efficient, if the worker has the ability to run
```

Comment:
Yes, that's why I suggested to use math/big. Normal ints won't be able to handle long candidate passwords. We could use int64, it'll be enough for this project. But for passwords of length 10ish or more then you need a bignum library.